### PR TITLE
3 get topics

### PR DESCRIPTION
**Pull request by alicjafleszar**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const express = require('express')
+const topicsRouter = require('./topics')
+
+const app = express()
+app.use(express.json())
+
+app.use('/api/topics', topicsRouter)
+
+app.use((req, res, next) => {
```
Comment (B7Bprog) [inline app.js]: Here as a first parameter you could add '*' to catch all invalid paths. 

An even better way is to use app.all(). (app.all('*', ()=>{}).

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+beforeEach(() => seed(testData))
+afterAll(() => db.end())
+
+describe('/api/topics', () => {
+    describe('GET - status 200 - responds with an array of topic objects', () => {
+        test('responds with an array of objects', () => {
+            return request(app)
+                .get('/api/topics')
+                .expect(200)
+                .then(({ body: { topics } }) => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:16]: Nice nested destructuring here.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                    expect(topics).toHaveLength(3)
+                    topics.forEach(topic => expect(typeof topic).toBe('object'))
+                })
+        })
+        test('each object in returned array has `slug` and `description` properties', () => {
+            return request(app)
+                .get('/api/topics')
+                .expect(200)
+                .then(({ body: { topics } }) => {
+                    topics.forEach(topic => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:26]: This is all good, but for the future I recommend the use of the following flexible object matchers:

.toMatchObject
.objectContaining

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Alicja, please see my comments, refactor your code, then it's good to merge. 👍

**Review by Rayhaan2k9 — APPROVED** [review]
Well done Alicja! Merge away 😁

**Pull request closed without merging**